src/api/local-api.tsx

```diff
@@ -56,6 +58,28 @@ export class DatacoreLocalApi {
         return this.api.core;
     }
 
+    //////////////////////////////
+    // Script loading utilities //
+    //////////////////////////////
+
+    // Note: Script loading is a bit jank, since it has to be asynchronous due to IO (unless of course we wanted to cache
+    // every single script in the vault in memory, which seems terrible for performance). It functions by essentially
+    // returning a lazy proxy.
+
+    /**
+     * Asynchronously load a javascript block from the given path or link; this method supports loading code blocks
+     * from markdown files via the link option
+     *
+     */
+
+    public readonly scriptCache: ScriptCache = new ScriptCache(this.core.datastore);
```

Comment:
I would create this in the constructor or move it to the top with the other variables.